install/kubernetes/cilium/templates/cilium-operator-clusterrole.yaml

```diff
@@ -43,6 +43,22 @@ rules:
   - get
   - list
   - watch
+- apiGroups:
+  - ""
+  resources:
+  # to perform LoadBalaner IP allocation for BGP
+  - services
+  verbs:
+  - get
+  - list
+  - watch
+  - update
```

Comment:
Hmm, good call out. The reason I added it is because MetalLB has code to [update](https://github.com/metallb/metallb/blob/124f6beb9165078111de9bd32d811c4e94784492/controller/main.go#L80) the object . However, if you follow that code path, there's nothing that MetalLB actually changes in the service's annotations or spec. So upstream MetalLB doesn't need this permission at all either. Upstream PR: https://github.com/metallb/metallb/pull/823

Reply:
Updated fork PR to remove it and removed it from cilium-operator clusterrole.